application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationTisStatusUpdateService.java

```diff
@@ -0,0 +1,87 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.service;
+
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.APPROVED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.REJECTED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationStatus.COMPLETED;
+
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.revalidation.dto.RecommendationStatusCheckDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+import uk.nhs.hee.tis.revalidation.entity.Recommendation;
+import uk.nhs.hee.tis.revalidation.repository.DoctorsForDBRepository;
+import uk.nhs.hee.tis.revalidation.repository.RecommendationRepository;
+
+@Slf4j
+@Service
+public class RecommendationTisStatusUpdateService {
+
+  @Autowired
+  private RecommendationRepository recommendationRepository;
+  @Autowired
+  private DoctorsForDBRepository doctorsForDBRepository;
+
+  @Value("${app.rabbit.reval.exchange.recommendationstatuscheck}")
+  private String revalExchangeRecommendationStatus;
+
+  @RabbitListener(queues = "${reval.queue.recommendationstatuscheck.update.recommendation}")
+  public void receiveMessageForRecommendationStatusUpdate(
+      final RecommendationStatusCheckDto recommendationStatusCheckDto) {
+
+    String gmcReferenceNumber = recommendationStatusCheckDto.getGmcReferenceNumber();
+    Optional<DoctorsForDB> optionalDoctorsForDB = doctorsForDBRepository
+        .findById(gmcReferenceNumber);
+    Optional<Recommendation> optionalRecommendation = recommendationRepository
+        .findById(recommendationStatusCheckDto.getRecommendationId());
+
+    if (optionalRecommendation.isPresent() && optionalRecommendation.get().getGmcNumber()
+        .equals(gmcReferenceNumber)) {
+      final var recommendation = optionalRecommendation.get();
+
+      //if gmc outcome is approved, update the relevant recommendation to approved, tis status to complete
+      if (recommendationStatusCheckDto.getOutcome().name().equals(APPROVED.name())) {
```

Comment:
If you store recommendationStatusCheckDto.getOutcome() in a variable you could combine these two if statements into:

`if(outcome..name().equals(APPROVED.name()) || outcome.name().equals(Rejected.name()) {`
` recommendation.setOutcome(outcome);`
       `recommendation.setRecommendationStatus(COMPLETED);`
`}`

Reply:
Nice thought

Reply:
I have refactored the code. Thanks